include/GafferScene/Capsule.h

```diff
@@ -58,11 +58,21 @@ class GAFFERSCENE_API Capsule : public IECoreScenePreview::Procedural
 		/// caller to provide a `hash` that uniquely identifies the entire
 		/// subtree from the root down, taking into account the context.
 		///
-		/// The capsule is invalidated by any subsequent graph edits
-		/// that dirty the scene (because the stored hash will no longer
-		/// match the scene).  We rely on the node creating the capsule
-		/// to flag it as dirty, and trigger a cancellation that will
-		/// prevent the invalid capsule from being used.
+		/// The capsule is invalidated by any subsequent graph edits that
+		/// modify the scene below `root`. Usage of an invalidated capsule
+		/// is undefined behaviour. In practice, nodes that create capsules
+		/// avoid such usage in two ways :
+		///
+		/// 1. Before any node graph edit is made, all existing processes
+		///    performing computes are cancelled. This prevents renderers
+		///    from continuing to use the capsule.
+		/// 2. After the node graph is edited, the capsule-generating
+		///    node generates a new `objectHash()` so that any new processes
+		///    will retrieve a new capsule from the node.
```

Comment:
You've removed the phrase "flag it as dirty", but that is also necessary, isn't it?  I guess that is kind of implicit in "generate a new `objectHash()`", but in order to do that, the node must both have an objectHash() which will return a new value, and also ensure that affects() is implemented correctly so that the hash gets dirtied?  ( Maybe I'm being too pendantic here )

Reply:
> You've removed the phrase "flag it as dirty", but that is also necessary, isn't it?

It is necessary yes, because if a node's output changes in any way then it has to signal dirtiness at the same time. But I was trying to get away from what I saw as a bit of a contradiction between "flag it as dirty" and https://github.com/GafferHQ/gaffer/commit/02ac22ee8a5ea91716f3f9bb6f714198515d5648. In that commit, we stopped using the dirty count to track invalidation, because the scene (as a whole) can be dirtied, without the hash for the capsule necessarily changing (because the capsule takes a subset of the scene). That's a distinction you made in order to optimise capsule output from the Instancer. So I thought that focussing on changes to the capsule's hash (which implies dirtiness) was better than focussing on dirtiness (which may not necessitate a change of capsule hash).